MatrixSDK/Utils/Realm/RLMSupport.swift

```diff
@@ -0,0 +1,174 @@
+}
+
+extension _RLMDictionaryIterator where Self: RLMCollection {
+    /// :nodoc:
+    public func makeIterator() -> RLMDictionaryIterator {
+        return RLMDictionaryIterator(self)
+    }
+}
+
+// Sequence conformance for RLMArray, RLMDictionary, RLMSet and RLMResults is provided by RLMCollection's
+// `makeIterator()` implementation.
+extension RLMArray: Sequence, _RLMCollectionIterator { }
+extension RLMDictionary: Sequence, _RLMDictionaryIterator {}
+extension RLMSet: Sequence, _RLMCollectionIterator {}
+extension RLMResults: Sequence, _RLMCollectionIterator {}
+
+/**
+ This struct enables sequence-style enumeration for RLMObjects in Swift via `RLMCollection.makeIterator`
+ */
+public struct RLMCollectionIterator: IteratorProtocol {
+    private var iteratorBase: NSFastEnumerationIterator
+
+    internal init(_ collection: RLMCollection) {
+        iteratorBase = NSFastEnumerationIterator(collection)
+    }
+
+    public mutating func next() -> RLMObject? {
+        return iteratorBase.next() as! RLMObject?
+    }
+}
+
+/**
+ This struct enables sequence-style enumeration for RLMDictionary in Swift via `RLMDictionary.makeIterator`
+ */
+public struct RLMDictionaryIterator: IteratorProtocol {
+    private var iteratorBase: NSFastEnumerationIterator
+    private let dictionary: RLMDictionary<AnyObject, AnyObject>
+
+    internal init(_ collection: RLMCollection) {
+        dictionary = collection as! RLMDictionary<AnyObject, AnyObject>
+        iteratorBase = NSFastEnumerationIterator(collection)
+    }
+
+    public mutating func next() -> RLMDictionarySingleEntry? {
+        let key = iteratorBase.next()
+        if let key = key {
+            return (key: key as Any, value: dictionary[key as AnyObject]) as? RLMDictionarySingleEntry
+        }
+        if key != nil {
+            fatalError("unsupported key type")
+        }
+        return nil
+    }
+}
+
+// Swift query convenience functions
+extension RLMCollection {
+    /**
+     Returns the index of the first object in the collection matching the predicate.
+     */
+    public func indexOfObject(where predicateFormat: String, _ args: CVarArg...) -> UInt {
+        guard let index = indexOfObject?(with: NSPredicate(format: predicateFormat, arguments: getVaList(args))) else {
+            fatalError("This RLMCollection does not support indexOfObject(where:)")
+        }
+        return index
+    }
+
+    /**
+     Returns all objects matching the given predicate in the collection.
+     */
+    public func objects(where predicateFormat: String, _ args: CVarArg...) -> RLMResults<NSObject> {
+        return objects(with: NSPredicate(format: predicateFormat, arguments: getVaList(args))) as! RLMResults<NSObject>
+    }
+}
+
+extension RLMCollection {
+    /// Allows for subscript support with RLMDictionary.
+    public subscript(_ key: String) -> AnyObject? {
+        get {
+            (self as! RLMDictionary<NSString, AnyObject>).object(forKey: key as NSString)
```

Comment:
Would `as?` be safer here given that the return value is an optional anyway?

Reply:
In fact this file come from Realm company. That's why I didn't touch it. But I agree with your comment.

Reply:
Ohh, sorry I missed the copyright comment at the top. 🤦‍♂️

All good then.